src/main/java/com/minecolonies/coremod/client/gui/ViewFilterableList.java

```diff
@@ -185,6 +185,17 @@ private void updateResources()
         allItems.clear();
         allItems.addAll(getBlockList(filterPredicate));
         allItems.addAll(getExceptions().stream().filter(storage -> filterPredicate.test(storage.getItemStack())).collect(Collectors.toList()));
+
+        allItems.sort((o1, o2) -> {
+            if(!building.isAllowedItem(id, new ItemStorage(o1.getItemStack())) && building.isAllowedItem(id, new ItemStorage(o2.getItemStack()))){
```

Comment:
should probably store the isAllowedItem check in a boolean value to not query it twice in the else if case

Reply:
Agreed